torch/onnx/symbolic_opset9.py

```diff
@@ -529,6 +529,25 @@ def reciprocal(g: jit_utils.GraphContext, self):
 @_beartype.beartype
 def cat(g: jit_utils.GraphContext, tensor_list, dim):
     tensors = symbolic_helper._unpack_list(tensor_list)
+    # torch.cat ignores empty tensors such as `torch.Tensor([])`
+    # These needs to be removed as input from ONNX's concat too, otherwise shape inference
+    # will likely fail due to inputs with different ranks (0 for empty tensor, > 0 for anything else)
+    nonempty_tensors = []
+    for t in tensors:
+        if symbolic_helper._is_constant(t) and not symbolic_helper._get_tensor_dim_size(t, 0):
```

Comment:
Should tensors with 0 in their dimension size count as "empty tensor" under this context? For example `torch.ones(2, 0, 3)`. (Maybe create a helper function `is_empty_tensor`)

Reply:
That is a good question. How would we detect one scenario or the other? both would have value `0`